x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/__mocks__/test_adapters.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { responseMock } from './response_factory';
+
+type ResponseMock = ReturnType<typeof responseMock.create>;
+type Method = keyof ResponseMock;
+
+type MockCall = any; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+interface ResponseCall {
+  body: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  status: number;
+}
+
+interface Response extends ResponseCall {
+  calls: ResponseCall[];
+}
+
+const buildResponses = (method: Method, calls: MockCall[]): ResponseCall[] => {
+  if (!calls.length) return [];
+
+  switch (method) {
+    case 'ok':
+      return calls.map(([call]) => ({ status: 200, body: call?.body }));
+    case 'custom':
+      return calls.map(([call]) => ({
+        status: call?.statusCode,
+        body: JSON.parse(call?.body),
+      }));
+    default:
+      throw new Error(`Encountered unexpected call to response.${method}`);
+  }
+};
+
+export const responseAdapter = (response: ResponseMock): Response => {
+  const methods = Object.keys(response) as Method[];
+  const calls = methods
+    .reduce((responses, method) => {
+      const methodMock = response[method];
+      return [...responses, ...buildResponses(method, methodMock.mock.calls)];
+    }, [] as Response['calls'])
+    .sort((call, other) => other.status - call.status);
```

Comment:
You can use reduce template here to avoid the as cast and that will be how TypeScript prefers it.
```ts
  const calls = methods
    .reduce<Response['calls']>((responses, method) => {
      const methodMock = response[method];
      return [...responses, ...buildResponses(method, methodMock.mock.calls)];
    }, [])
    .sort((call, other) => other.status - call.status);
```

Reply:
6a84e23e1aa20ff7d0812bb28f440bd63a23fad7